tests/unit/session/test_parallel.py

```diff
@@ -189,3 +189,44 @@ def test_parallel_show_output(cmd, initproj, monkeypatch):
     assert "stderr env" not in result.out, result.output()
     assert "stdout always" in result.out, result.output()
     assert "stderr always" in result.out, result.output()
+
+
+def test_parallel_result_json(cmd, initproj, tmpdir):
+    import json
+
+    initproj(
```

Comment:
Can you test the parallel case we were speaking above? just use threads and enqueue another parallel result json request on a only 👌

Reply:
Done. This only works if the virtualenvs already exist and don't change (I have a part for that in the test), otherwise the virtualenv calls will conflict. I guess you should add a lock for virtualenv creation.

Reply:
Indeed, we can do it in a follow-up pr, for now just call it first without any result json to create them, and run the test in a follow-up.